src/main.lua

```diff
@@ -386,47 +383,53 @@ addEventingHandler(ActionMap.VaultedTransfer, utils.hasMatchingTag("Action", Act
 		return
 	end
 
-	local result, err = balances.vaultedTransfer(
-		msg.From,
-		msg.Tags.Recipient,
-		tonumber(msg.Tags.Quantity),
-		tonumber(msg.Tags["Lock-Length"]),
-		msg.Timestamp,
-		msg.Id
-	)
+	local from = utils.formatAddress(msg.From)
+	local recipient = utils.formatAddress(msg.Tags.Recipient)
+	local quantity = tonumber(msg.Tags.Quantity)
+	local lockLengthMs = tonumber(msg.Tags["Lock-Length"])
+	local timestamp = tonumber(msg.Timestamp)
+	local msgId = msg.Id
+
+	local result, err = vaults.vaultedTransfer(from, recipient, quantity, lockLengthMs, timestamp, msgId)
+	print("Created vault" .. json.encode(Vaults[recipient]))
 	if err then
 		ao.send({
 			Target = msg.From,
 			Tags = { Action = "Invalid-Vaulted-Transfer", Error = "Invalid-Vaulted-Transfer" },
 			Data = tostring(err),
 		})
 	else
+		-- sender gets an immediate credit notice
 		ao.send({
-			Target = msg.From,
-			Recipient = msg.Tags.Recipient,
-			Quantity = msg.Tags.Quantity,
-			Tags = { Action = "Debit-Notice" },
+			Target = from,
+			Recipient = recipient,
+			Quantity = quantity,
+			Tags = { Action = "Credit-Notice", ["Vault-Id"] = msgId },
```

Comment:
Why did we move away from a Debit-Notice here?

Reply:
bc i'm a fool and got the definitions mixed up in my head - reverted back in [7d48ad4](https://github.com/ar-io/ar-io-network-process/pull/102/commits/7d48ad4516cf276782179f9b0f8805de4362ec1a)